libraries/AP_GPS/AP_GPS_UBLOX.cpp

```diff
@@ -962,6 +1163,28 @@ AP_GPS_UBLOX::_parse_gps(void)
                 if (_hardware_generation != UBLOX_F9) {
                     // need to ensure time mode is correctly setup on F9
                     _unconfigured_messages |= CONFIG_TMODE_MODE;
+
+                    // see if RTK moving baseline base is wanted
+                    if (role == AP_GPS::GPS_ROLE_MB_BASE) {
+                        rtcm3_parser = new RTCM3_Parser;
+                        if (rtcm3_parser == nullptr) {
+                            GCS_SEND_TEXT(MAV_SEVERITY_ERROR,
+                                          "GPS: u-blox %d MB Base FAILED NO MEMORY",
+                                          state.instance + 1);
+                        } else {
+                            GCS_SEND_TEXT(MAV_SEVERITY_INFO,
+                                          "GPS: u-blox %d config start MB Base",
+                                          state.instance + 1);
+                        }
+                        _unconfigured_messages |= CONFIG_RTK_MOVBASE;
```

Comment:
I really want a predictable order to the configuration. This stuff is hard enough to get right without us first getting the device into a known state with the base config before we start using the advanced config.

Reply:
All you have to do is set the `STEP_RTK_MOVBASE` after `STEP_VERSION`. It should be there anyway since you can't do it till you know the version. If that's set the order is absolutely predictable, as we won't do this until after the hardware version is printed out. The driver is actually very predictable for how the configuration process works. By separating stuff out like this it actually gets harder to predict the order, so the comment stands. (The order you have it in actually makes it take longer for us to correctly have configured the GPS's on the system, and thus get everything into a working setup, as it requires us to rerun many of the config steps).

Reply:
yep, that is a lot better, sorry for being slow on this. Fixed to move STEP_RTK_MOVBASE after STEP_VERSION and config is now faster